app/src/main/java/org/eyeseetea/malariacare/domain/usecase/pull/DownloadMediaUseCase.java

```diff
@@ -46,44 +49,68 @@ public void execute(Callback callback) {
 
     @Override
     public void run() {
-        List<String> uids = getResourceUrlsToDownload();
+        System.out.println("Running DownloadMediaUseCase");
+        IFileDownloader.Callback callback = new IFileDownloader.Callback() {
+            @Override
+            public void onError(FileDownloadException ex) {
+                mCallback.onError(ex);
+            }
+
+            @Override
+            public void onSuccess(List<Media> syncMedias) {
+                int numOfSyncedFiles = saveDownloadedMedia(syncMedias);
+                List<Media> allMedias = mMediaRepository.getAll();
+                removeNotDownloadedMedia(syncMedias, allMedias);
+                mCallback.onSuccess(numOfSyncedFiles);
+            }
+
+        };
+
+        List<Media> medias = mFileDownloader.init(
```

Comment:
I think that returns previously medias is not necessary and init method either. We have to download simply all medias from drive folder and pass DriveRootFolderUid and ProgramUId in the download method of FileDownloader